Source/WebCore/dom/Document.cpp

```diff
@@ -6841,8 +6841,9 @@ void Document::initSecurityContext()
         return;
     }
 
-    contentSecurityPolicy()->copyStateFrom(ownerFrame->document()->contentSecurityPolicy());
-    contentSecurityPolicy()->updateSourceSelf(ownerFrame->document()->securityOrigin());
+    CheckedPtr contentSecurityPolicy = this->contentSecurityPolicy();
+    contentSecurityPolicy->copyStateFrom(ownerFrame->protectedDocument()->checkedContentSecurityPolicy().get());
+    contentSecurityPolicy->updateSourceSelf(ownerFrame->document()->securityOrigin());
```

Comment:
Why don't we want `protectedDocument()` on this second line?

Reply:
Protecting the document here doesn't protect anything, it is the security origin we would need to protect since it is a ref counted object. I'll switch to `protectedSecurityOrigin()`.